plugins/nanospell/plugin.js

```diff
@@ -313,28 +327,33 @@
 
 			/* #2 setup layer */
 			/* #3 nanospell util layer */
-			var start = function () {
+			function start() {
 				editor.getCommand('nanospell').setState(CKEDITOR.TRISTATE_ON);
 				state = true;
-				var words = getWords(editor.document.$.body, maxRequest);
-				if (words.length == 0) {
-					render();
-				} else {
-					send(words);
-				}
-			};
-			var stop = function () {
+
+				startSpellCheckTimer(50);
+			}
+
+			function stop() {
 				editor.getCommand('nanospell').setState(CKEDITOR.TRISTATE_OFF);
 				state = false;
 				clearAllSpellCheckingSpans(editor.editable());
-			};
+			}
 
 			function checkNow() {
-				if (!selectionCollapsed()) {
+				if (!selectionCollapsed() || self._isBusy) {
 					return;
 				}
 				if (state) {
-					start();
+
```

Comment:
It just stores whether the spellcheck plugin is active.  (you can toggle it on/off if you add a button.)  I haven't cleaned it up but I think it should probably be unified with the CKEditor command state concept.  (looks like they originally kept them side by side and tried to keep them in sync)